modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -488,31 +590,92 @@ namespace ie {
+            break;
+        }
         case cv::GMetaArg::index_of<cv::GFrameDesc>():
-            {
-                const auto &meta = util::get<cv::GFrameDesc>(mm);
-                switch (meta.fmt) {
-                    case cv::MediaFormat::NV12:
-                        ii->getPreProcess().setColorFormat(IE::ColorFormat::NV12);
-                        break;
-                    case cv::MediaFormat::BGR:
-                        // NB: Do nothing
-                        break;
-                    default:
-                        GAPI_Assert(false && "Unsupported media format for IE backend");
-                }
-                ii->setPrecision(toIE(CV_8U));
-                break;
+        {
+            const auto &meta = util::get<cv::GFrameDesc>(mm);
+            switch (meta.fmt) {
+                case cv::MediaFormat::NV12:
+                    ii->getPreProcess().setColorFormat(IE::ColorFormat::NV12);
+                    break;
+                case cv::MediaFormat::BGR:
+                    // NB: Do nothing
+                    break;
+                default:
+                    GAPI_Assert(false && "Unsupported media format for IE backend");
             }
+            ii->setPrecision(toIE(CV_8U));
+            break;
+        }
         default:
             util::throw_error(std::runtime_error("Unsupported input meta for IE backend"));
     }
 }
 
+struct PostOutputs
+{
+    // NB: Should be const to pass into SetCompletionCallback
+    void operator()() const
+    {
+       for (auto i : ade::util::iota(uu.params.num_out))
+       {
+            auto& out_mat = ctx->outMatR(i);
+            IE::Blob::Ptr this_blob = iec.this_request.GetBlob(uu.params.output_names[i]);
+            copyFromIE(this_blob, out_mat);
+            out.post(ctx->out(i));
+       }
+
+        sync.release_and_notify();
+    }
+
+    IECompiled                            &iec ;
+    const IEUnit                          &uu  ;
+    cv::gimpl::GIslandExecutable::IOutput &out ;
+    cv::gimpl::ie::SyncPrim               &sync;
+    std::shared_ptr<IECallContext>         ctx ;
+};
+
+struct PostOutputsList
+{
+    // NB: Should be const to pass into SetCompletionCallback
+    void operator()() const
+    {
+        for (auto i : ade::util::iota(uu.params.num_out)) {
+            std::vector<cv::Mat> &out_vec = ctx->outVecR<cv::Mat>(i);
+
+            IE::Blob::Ptr out_blob = iec.this_request.GetBlob(uu.params.output_names[i]);
+
+            cv::Mat out_mat(cached_dims[i], toCV(out_blob->getTensorDesc().getPrecision()));
+            // FIXME: Avoid data copy. Not sure if it is possible though
+            copyFromIE(out_blob, out_mat);
+            out_vec.push_back(std::move(out_mat));
+        }
+        // NB: Callbacks run synchronously yet, so the lock isn't necessary
+        auto&& nouts = ctx->outVecR<cv::Mat>(0).size();
+        // NB: Now output vector is collected and can be posted to output
+        if (nrequests == nouts) {
```

Comment:
`nouts` is read as `number of outputs` but this semantic is wrong.

This is clearly the output vector _size_.

Reply:
Fixed